pkg/util/typeutil/schema.go

```diff
@@ -1519,60 +1536,57 @@ type sparseFloatVectorJSONRepresentation struct {
 }
 
 // accepted format:
-//   - {"indices": [1, 2, 3], "values": [0.1, 0.2, 0.3]}
-//   - {"1": 0.1, "2": 0.2, "3": 0.3}
+//   - {"indices": [1, 2, 3], "values": [0.1, 0.2, 0.3]}    # format1
+//   - {"1": 0.1, "2": 0.2, "3": 0.3}                       # format2
 //
 // we don't require the indices to be sorted from user input, but the returned
 // byte representation must have indices sorted
-func CreateSparseFloatRowFromJSON(input []byte) ([]byte, error) {
+func CreateSparseFloatRowFromJSON(input map[string]interface{}) ([]byte, error) {
 	var indices []uint32
 	var values []float32
 
-	var vec sparseFloatVectorJSONRepresentation
-	decoder := json.NewDecoder(bytes.NewReader(input))
-	decoder.DisallowUnknownFields()
-	err := decoder.Decode(&vec)
-	if err == nil {
-		if len(vec.Indices) != len(vec.Values) {
-			return nil, fmt.Errorf("indices and values length mismatch")
-		}
-		if len(vec.Indices) == 0 {
-			return nil, fmt.Errorf("empty indices/values in JSON input")
-		}
-		indices = vec.Indices
-		values = vec.Values
-	} else {
-		var vec2 map[uint32]float32
-		decoder = json.NewDecoder(bytes.NewReader(input))
-		decoder.DisallowUnknownFields()
-		err = decoder.Decode(&vec2)
-		if err != nil {
-			return nil, fmt.Errorf("failed to parse JSON input: %v", err)
-		}
+	if len(input) == 0 {
+		return nil, fmt.Errorf("empty JSON input")
+	}
 
-		for idx, val := range vec2 {
-			indices = append(indices, idx)
-			values = append(values, val)
+	// try format1
+	indices, ok1 := input["indices"].([]uint32)
+	values, ok2 := input["values"].([]float32)
```

Comment:
what if the input is `{"indices": [1, 2, 3], "1": 0.1, "2": 0.2, "3": 0.3}`, I believe we require a more stringent inspection.

Reply:
we will return error for this mixed format